doc/user/content/sql/explain-analyze.md

```diff
@@ -0,0 +1,234 @@
+
+When optimizing a query, it helps to be able to attribute 'cost' to its parts,
+starting with how much time is spent computing in each part overall. For example, Materialize
+reports the time spent in each _dataflow operator_ in
+[`mz_introspection.mz_compute_operator_durations_histogram`](/sql/system-catalog/mz_introspection/#mz_compute_operator_durations_histogram).
+Using `EXPLAIN ANALYZE`, we can attribute the time spent in each operator to the higher-level, more
+intelligible LIR operators.
+
+{{< warning >}}
+`EXPLAIN` is not part of Materialize's stable interface and is not subject to
+our backwards compatibility guarantee. The syntax and output of `EXPLAIN` may
+change arbitrarily in future versions of Materialize.
+{{< /warning >}}
+
+## Syntax
+
+```mzsql
+EXPLAIN ANALYZE
+    [ HINTS | [<property> [, <property ...>] [WITH SKEW]]]
+    FOR [INDEX ... | MATERIALIZED VIEW ...]
+    [ AS SQL ]
+;
+```
+
+Here `<property>` is either `CPU` or `MEMORY`; you may list them in any order, but each may appear only once.
+
+### Explained object
+
+The following object types can be explained.
+
+Explained object | Description
+------|-----
+**INDEX name** | Display information for an existing index.
+**MATERIALIZED VIEW name** | Display information for an existing materialized view.
+
+### Properties to analyze
+
+
+Property | Description
+------|-----
+**`HINTS`** | Annotates the LIR plan with TopK hints.
+**`CPU`** | Annotates the LIR plan with information about CPU time consumed.
+**`MEMORY`** | Annotates the LIR plan with information about memory consumed.
+**`WITH SKEW`** | Adds additional information about average and per-worker consumption (of `CPU` or `MEMORY`).
+
+Note that `CPU` and `MEMORY` can be combined, separated by commas.
+When running `EXPLAIN ANALYZE` for `CPU` or `MEMORY`, you can specify `WITH SKEW` to observe average and per worker information, as well.
+
+## Details
+
+Under the hood, `EXPLAIN ANALYZE` runs SQL queries that correlate [`mz_introspection` performance information](https://materialize.com/docs/sql/system-catalog/mz_introspection/) with the LIR operators in [`mz_introspection.mz_lir_mapping`](../../sql/system-catalog/mz_introspection/#mz_lir_mapping).
+
+The attribution examples in this
+section reference the `wins_by_item` index (and the underlying `winning_bids`
+view) from the [quickstart
+guide](/get-started/quickstart/#step-2-create-the-source):
+
+```sql
+CREATE SOURCE auction_house
+FROM LOAD GENERATOR AUCTION
+(TICK INTERVAL '1s', AS OF 100000)
+FOR ALL TABLES;
+
+CREATE VIEW winning_bids AS
+  SELECT DISTINCT ON (a.id) b.*, a.item, a.seller
+    FROM auctions AS a
+    JOIN bids AS b
+      ON a.id = b.auction_id
+   WHERE b.bid_time < a.end_time
+     AND mz_now() >= a.end_time
+   ORDER BY a.id, b.amount DESC, b.bid_time, b.buyer;
+
+CREATE INDEX wins_by_item ON winning_bids (item);
```

Comment:
So ... running `explain analyze for index` works but if I update to create a materialized view and run `explain analyze for mat view`, I get lagging objects. Using one of the mzbuild from this pr.

Reply:
Added a patch for some docs tweak.  I left the transform-data/optimization page alone for now because that page needs a bit more rewrite.